vault/core.go

```diff
@@ -1866,48 +1901,50 @@ func (c *Core) runStandby(doneCh, stopCh, manualStepDownCh chan struct{}) {
 		}
 
 		// Monitor a loss of leadership
-		var manualStepDown bool
+		var stopOpts StopOptions
+		grabStateLock := true
 		select {
 		case <-leaderLostCh:
 			c.logger.Warn("core: leadership lost, stopping active operation")
-		case <-stopCh:
-			c.logger.Warn("core: stopping active operation")
+		case stopOpts = <-stopCh:
+			// This case comes from sealInternal; we will already be having the
+			// state lock held so we do toggle grabStateLock to false
+			grabStateLock = false
+			close(stopCh)
 		case <-manualStepDownCh:
 			c.logger.Warn("core: stepping down from active operation to standby")
 			manualStepDown = true
 		}
 
 		metrics.MeasureSince([]string{"core", "leadership_lost"}, activeTime)
 
-		// Clear ourself as leader
-		if err := c.clearLeader(uuid); err != nil {
-			c.logger.Error("core: clearing leader advertisement failed", "error", err)
-		}
-
 		// Tell any requests that know about this to stop
 		if c.activeContextCancelFunc != nil {
 			c.activeContextCancelFunc()
```

Comment:
I wonder if in the prepare secondary case we want to close the context?

Reply:
Yeah I think so, because we're sealing. Any outstanding request won't be able to do much.

Reply:
Okay then we'll have to make sure that nothing in the prepare secondary is using the active context. I'm not sure that anything is, just a thought...